Fix for CPAN RT # 32198, Debian Bug #499973.
Return early if there are no headers to retrieve.
David Bremner <bremner@unb.ca>
--- libemail-abstract-perl.orig/lib/Email/Abstract/MailInternet.pm
+++ libemail-abstract-perl/lib/Email/Abstract/MailInternet.pm
@@ -25,6 +25,12 @@
     my ($class, $obj, $header) = @_; 
     my @values = $obj->head->get($header); 
 
+    # early exit to avoid warnings about undefined values
+    # based on a patch by adriano ferreira, except we return
+    # an empty list in a list context.
+    if (!scalar(@values)){
+	return wantarray  ? () : undef;
+    }
     # No reason to s/// lots of values if we're just going to return one.
     $#values = 0 if not wantarray;
 
